### ISSUE TITLE
CustomImportOrder not always reports lexical order violations

### ISSUE THREAD

**Issue by ivanov-alex** [checkstyle/checkstyle#1469]
In example below I run ImportOrder and CustomImportOrder on one file.
It has 2 imports in a wrong lexical order. ImportOrder fires 2 violation (message starts with "Wrong order"), while CustomImportOrder fires only one (message starts with "Wrong lexicographical order").

CustomImportOrder should also fire violation for "com.*"

Configuration:
```xml
<?xml version="1.0"?>
<!DOCTYPE module PUBLIC "-//Puppy Crawl//DTD Check Configuration 1.3//EN" "http://www.puppycrawl.com/dtds/configuration_1_3.dtd">
<module name = "Checker">
    <property name="charset" value="UTF-8"/>   <property name="severity" value="error"/>
    <module name="TreeWalker">
    <module name="ImportOrder">
        <property name="groups" value="/^javax?\./"/>
        <property name="ordered" value="true"/>
        <property name="separated" value="false"/>
        <property name="option" value="top"/>
        <property name="sortStaticImportsAlphabetically" value="true"/>
    </module>
    <module name="CustomImportOrder">
        <property name="customImportOrderRules" value="STATIC###STANDARD_JAVA_PACKAGE"/>
        <property name="specialImportsRegExp" value="org"/>
        <property name="sortImportsInGroupAlphabetically" value="true"/>
        <property name="separateLineBetweenGroups" value="false"/>
    </module>
    </module>
</module>
```
Source file
```java
package java.util.concurrent;
import static java.awt.Button.ABORT;
import static javax.swing.WindowConstants.*;
import static java.io.File.createTempFile;
import java.util.*;
import java.util.StringTokenizer;
import com.puppycrawl.tools.*;
import com.*;
import org.apache.*;

public class InputCustomImportOrder_NoSeparator {
}
```
Output
```
Starting audit...
InputCustomImportOrder_NoSeparator.java:4: Wrong lexicographical order for 'java.io.File.createTempFile' import.
InputCustomImportOrder_NoSeparator.java:4: Wrong order for 'java.io.File.createTempFile' import.
InputCustomImportOrder_NoSeparator.java:8: Wrong order for 'com.*' import.
Audit done.
Checkstyle ends with 3 errors.
```

**Comment by romani** [comment]
how IDEs sort such case ? according to ASCII (https://en.wikipedia.org/wiki/ASCII#ASCII_printable_code_chart) "*" has lower code than any letter.

**Comment by ivanov-alex** [comment]
IDE also sorts them according to ASCII (* first)

**Comment by ivanov-alex** [comment]
One more problem with lexical ordering:

```java
1   package com.gc.materialdesign.views;
2   
3   import com.gc.materialdesign.R;
4   import com.gc.materialdesign.utils.Utils;
5   
6   import android.content.Context;
7   import android.graphics.Bitmap;
8   import android.graphics.Canvas;
9   import android.graphics.Color;
10  import android.graphics.Paint;
11  import android.graphics.Rect;
12  import android.graphics.Bitmap.Config;
13  import android.graphics.drawable.GradientDrawable;
14  import android.graphics.drawable.LayerDrawable;
15  import android.util.AttributeSet;
16  import android.view.MotionEvent;
17  import android.widget.TextView;
```
config:
```xml
        <module name="CustomImportOrder">
            <property name="specialImportsRegExp" value="com.google"/>
            <property name="sortImportsInGroupAlphabetically" value="true"/>
            <property name="customImportOrderRules" value="STATIC###SAME_PACKAGE(2)###THIRD_PARTY_PACKAGE###STANDARD_JAVA_PACKAGE"/>
        </module>
```
Output:
```
6: Wrong lexicographical order for 'android.content.Context' import.
12: Wrong lexicographical order for 'android.graphics.Bitmap.Config' import.
```
False positive for 'android.content.Context'. There should be no violation as this is first import in the group.

**Comment by romani** [comment]
merged
